CMakeLists.txt

```diff
@@ -63,42 +63,44 @@ find_library(LUA_LIBRARY lua)
 link_libraries(${LUA_LIBRARY})
 
 find_package(Doxygen QUIET)
+include_directories(include)
 
 set(NOVELRT_HEADERS
-  src/NovelWorldObject.h
-  src/GeoBounds.h
-  src/GeoVector.h
-  src/GraphicsCharacterRenderData.h
-  src/KeyCode.h
-  src/Lazy.h
-  src/NovelAudioService.h
-  src/NovelBasicFillRect.h
-  src/NovelBasicInteractionRect.h
-  src/NovelCommonArgs.h
-  src/NovelDebugService.h
-  src/NovelImageRect.h
-  src/NovelInteractionObject.h
-  src/NovelInteractionService.h
-  src/NovelLayeringService.h
-  src/NovelLoggingService.h
-  src/NovelObjectSortComparison.h
-  src/NovelRenderingService.h
-  src/NovelRenderObject.h
-  src/NovelUtilities.h
-  src/NovelRunner.h
-  src/NovelStepTimer.h
-  src/NovelTextRect.h
-  src/RGBAConfig.h
-  src/wrapper.hpp
-  src/NovelCamera.h
-  src/GeoMatrix4.h
-  src/CameraViewChangedEventArgs.h
-  src/ShaderProgram.h
-  src/CameraBlock.h
-  src/NovelRT.h)
+  include/NovelRT/Transform.h
+  include/NovelRT/Maths/GeoBounds.h
+  include/NovelRT/Maths/GeoVector.h
+  include/NovelRT/Graphics/GraphicsCharacterRenderData.h
+  include/NovelRT/Input/KeyCode.h
+  include/NovelRT/Utilities/Lazy.h
+  include/NovelRT/Audio/AudioService.h
+  include/NovelRT/Graphics/BasicFillRect.h
+  include/NovelRT/Input/BasicInteractionRect.h
+  include/NovelRT/Utilities/CommonArgs.h
+  include/NovelRT/DebugService.h
+  include/NovelRT/Graphics/ImageRect.h
+  include/NovelRT/Input/InteractionObject.h
+  include/NovelRT/Input/InteractionService.h
+  include/NovelRT/LayeringService.h
+  include/NovelRT/LoggingService.h
+  include/NovelRT/TransformSortComparison.h
+  include/NovelRT/Graphics/RenderingService.h
+  include/NovelRT/Graphics/RenderObject.h
+  include/NovelRT/Utilities/Misc.h
+  include/NovelRT/NovelRunner.h
+  include/NovelRT/Timing/StepTimer.h
+  include/NovelRT/Graphics/TextRect.h
+  include/NovelRT/Graphics/RGBAConfig.h
+  include/NovelRT/Graphics/Camera.h
+  include/NovelRT/Maths/GeoMatrix4.h
+  include/NovelRT/Graphics/CameraViewChangedEventArgs.h
+  include/NovelRT/Graphics/ShaderProgram.h
+  include/NovelRT/Graphics/CameraBlock.h
+  include/NovelRT/Windowing/WindowingService.h
+  include/NovelRT/Input/KeyState.h
+  include/NovelRT.h)
 
 set(NOVELRT_SOURCES
-        src/NovelWorldObject.cpp
+  src/NovelWorldObject.cpp
```

Comment:
nit: sources folder structure should mirror the include folder structure

Reply:
this is on my todo already. 😄